### ISSUE TITLE
Use fieldset and legend for grouped form inputs

### ISSUE THREAD

**Issue by taysea** [grommet/grommet#4108]
For grouped form components, such as multiple checkboxes or radiobuttons, they should be wrapped in a `<fieldset>` and `<legend>` for accessibility as opposed to a `<label>` as is currently happening. Each individual checkbox and radiobutton already has a `<label>`, so wrapping in a FormField just causes an extra label to be added to the DOM that isn't associated with an input.

Proposing that we either create a new component called "FormFieldGroup" that would wrap grouped inputs. Instead of applying a label, it would apply a legend and wrap everything in a fieldset.

### Expected Behavior

When grouped, form inputs should be wrapped in a `<fieldset>` and`<legend>`. Individual checkboxes/radiobuttons already have their own `<label>`. 

Resource for expected DOM output: https://www.w3schools.com/tags/tag_fieldset.asp
This is being done in other frameworks to achieve the correct DOM output: https://material-ui.com/components/checkboxes/#checkboxes-with-formgroup

### Actual Behavior

A checkboxgroup or radiobutton group is being wrapped with another `<label>` which ends up being orphaned in the DOM because it is not associated with any input.

### URL, screen shot, or Codepen exhibiting the issue

When 'FormField' label is not properly associated with any of the inputs because each checkbox already has its own label.

<img width="406" alt="Screen Shot 2020-05-22 at 10 18 32 AM" src="https://user-images.githubusercontent.com/12522275/82693065-ab469f80-9c15-11ea-84ba-a68038a0a44d.png">

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: stable
- Browser Name and version: chrome
- Operating System and version (desktop or mobile): macos

**Comment by ShimiSun** [comment]
Talking it through, here is the desired optimization for the code:

```
// In grommet
<FormField legend="Server Types">
  <CheckBoxGroup options={options} />
</FormField>

// in the DOM:
<fieldset>
   <legend>Server Types</legend>
   {...checkboxes}
</fieldset>
```
So to simplify it with pseudo:
```
if any groupComponent
   use <fieldset> & <legend>
else
   use label.
```

My only concern is that we currently don't use `<fieldset>` in the DOM and adding it might need to be thought through a little more.

**Comment by ShimiSun** [comment]
The bigger issue here is what role would `fieldset` have in grommet's Form and FormField, resolving it generically will be better than patching it for this single use case.

**Comment by taysea** [comment]
Looking back at this issue a little, was able to spin up a local example where I applied:
```
<FormField as="fieldset">
   <CheckBoxGroup />
</FormField>
```
This appropriately made the wrapper of the CheckBoxGroup a fieldset, but now we need to provide the user a way to provide the `legend` value. Thinking one of two things:
1. We add a `legend` prop to FormField to align with the DOM tag we want to insert. We would document that this should be used when a user is using `as="fieldset"`
2. We check internally in grommet: if user provides `as="fieldset"` and a `label` --> converted the label to legend. (this feels a little unintuitive)

Additional notes: `fieldset` has some default padding, margin, and border that we would want to strip out so it doesn't conflict with formfield styling.

**Comment by MikeKingdom** [comment]
Changing the FormField to render as a `fieldset` and `legend` would require styling from the theme to make the legend look like a regular formfield label. Since the goal is accessibility more than styling, I think a better solution would be to change CheckBoxGroup and RadioButtonGroup to set `role="group"` and `role="radiogroup"` respectively on the `div` containing the group contents. The you could use `aria-labelledby` on the group to point to the `FormField`. This seems to work well with NVDA (I didn't try with JAWS) and is one of the recommended ways to do accessible groups of radio buttons or checkboxes. The screen reader properly reads the FormField label and checkbox labels as you navigate to them or change them. For example
```
<FormField id="formfield_id" label="Location">
  <CheckBoxGroup aria-labelledby="formfield_id" options={["Maui", "Jerusalem", "Wuhan"]} />
</FormField>
```

We could make a convenience property to avoid the literal aria-labelledby attribute or find a way to figure it automatically.

Making a separate `<FieldSet legend="..."/>` component may be worthwhile to address situations like grouped input fields with a label like the `Shipping` and `Contact information` groups[ in this form](https://design-system.hpe.design/templates/forms#shipping). In this case you want some styling of the label for the group and maybe the `disabled` attribute available on a fieldset. You would also get the accessibility benefit from this (e.g. a screen reader would include the legend when talking about individual inputs in the fieldset.) This seems independent of the checkboxgroup situation since you wouldn't want styling different than a FormField in the checkboxgroup situation.
